net/grpc/grpc_resolver.go

```diff
@@ -0,0 +1,105 @@
+package grpc
+
+import (
+	"context"
+	"time"
+	"log"
+
+	"google.golang.org/grpc/resolver"
+
+	net_balancer "github.com/trafficstars/registry/net/balancer"
+)
+
+const defaultRefreshInterval = time.Second * 5
+
+type grpcMetadata struct {
+	serviceName          string
+	backend              *net_balancer.Backend
+	balancer             net_balancer.Balancer
+	maxRequestsByBackend int
+}
+
+type grpcResolver struct {
+	// Service name in the discovery registry
+	serviceName string
+
+	// Maximal amount of requests by backend
+	maxRequestsByBackend int
+
+	// Backend used at the moment
+	currentBackend *net_balancer.Backend
+
+	// Default connection balancer
+	balancer net_balancer.Balancer
+
+	// Refresh timer interval
+	freq time.Duration
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	cc     resolver.ClientConn
+
+	t *time.Ticker
+}
+
+// ResolveNow invoke an immediate resolution of the target that this dnsResolver watches.
+func (r *grpcResolver) ResolveNow(opt resolver.ResolveNowOption) {
+	r.refreshConnection()
+}
+
+// Close closes the dnsResolver.
+func (r *grpcResolver) Close() {
+	r.cancel()
+	if r.t != nil {
```

Comment:
IMHO, this is an unnecessary `if`. It only hides problem with `Close()`-ing non-initialized structure.

Reply:
Removed